docs/quick-start.md

```diff
@@ -0,0 +1,87 @@
+# Flaskerize Quick-Start Tutorial
+
+This guide is designed to get you up and running by showing you how to create a new Flask API using Flaskerize. We'll perform the following steps in this quick-start...
+
+1. Create a new folder and set up a virtual environment
+2. Install Flaskerize
+3. Use Flaskerize to create an initial Flask API
+4. Use Flaskerize to add an entity to your API
+
+## Pre-Requisites and Assumptions
+
+It's assumed that you have Python 3.7 installed. The illustrations in this tutorial assume that you're using a bash terminal, and that you're comfortable using a command line.
+
+OK....let's start...
+
+## Step 1 - Create a New Folder and Set Up a Virtual Environment
+
+Let's assume you're started from nothing. First, let's create a folder for your Flask API to live in.
+
+```bash
+mkdir flaskerize-example
+cd flaskertize-example
+```
+
+Now, let's set up a virtual environment for our API project, activate it, and then upgrade `pip` within that environment.
+
+```bash
+python -m venv venv
+source venv/bin/activate
+pip install --upgrade pip
+```
+
+## Step 2 - Install Flaskerize
+
+We're now ready to install Flaskeriez. Let's use `pip` to do just that...
+
+```bash
+pip install --upgrade pip
+```
+
+Once this command has completed you will have installed Flaskerize along with its dependencies. If you want to see the packages that were installed, run the following command:
+
+```bash
+pip list
+```
+
+This should show you something like this...
+
+``` bash
+$ pip list
+Package      Version
+------------ -------
+appdirs      1.4.3
+Click        7.0
+Flask        1.1.1
+flaskerize   0.10.0
+fs           2.4.11
+itsdangerous 1.1.0
+Jinja2       2.10.1
+MarkupSafe   1.1.1
+pip          19.2.3
+pytz         2019.2
+setuptools   40.8.0
+six          1.12.0
+termcolor    1.1.0
+Werkzeug     0.15.6
+```
+
+> note: the exact version numbers shown here may differ
+>
+## Step 3 - Use Flaskerize to create an initial Flask API
```

Comment:
This is great. One of the things that the Entity schematic does not currently do but I would like to add is have it self-register the newly created Entity within the API so that after generation you just refresh the API and the new API is available with functioning tests. Assuming the project follows the patterns I have proposed previously, this is mechanically only a couple of project modifications. However, the topic of modifying source code goes much deeper and is something I have thought about a lot and had some discussions with developers of the Angular CLI on. The solution seems to be to  round-trip Abstract Syntax Tree (AST) in order to properly rewrite code.

Until that functionality is there, it would be the icing on the cake in the getting-started section to show what steps are necessary to wire up the new entity. See [here](https://github.com/apryor6/flask_api_example) for a reference project. Basically you just need to add a single line to the routes.py like [this](https://github.com/apryor6/flask_api_example/blob/master/app/routes.py)

Reply:
My suggestion is to write any docs with all of the manual steps included. The docs can then serve as a good reference for any code improvements that should follow (as the improvements should aim to drive down the number of steps, or amount of manual code editing required). Does that sound OK?

Reply:
Completely agree.

Reply:
I'm going to create an Issue (if that's OK with you) with an outline of what I'll pick up in this PR. The essence will be "get the basics of some docs in place so that others can contribute". 

One of the questions I had was whether you wanted some basic Markdown docs for now, or would you like me to create Sphinx doc that could then be hosted on readthedocs.org?

I don't want to bite off more than I can complete (or do something you're not keen to accept as a PR), but equally...I think getting a readthedocs-setup in place would be a good thing longer term.

Reply:
Either is fine, but I do prefer the Sphinx docs as they can be hosted as "real" documentation. I'm willing to help out with that, and it is highly unlikely that there would be a reason for me to reject a documentation PR, so don't worry about that!